src/component/auth/Login.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+const Login = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState('');
+  const history = useHistory();
+
+  const handleLogin = async (e) => {
+    e.preventDefault();
+    try {
+      const response = await fetch("api/login", {
+        method: 'POST',
+        headers: {
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify({ email, password })
+      });
+
+      if (!response.ok) {
+        throw new Error('Login failed');
+      }
+
+      const result = await response.json();
+      localStorage.setItem("user-info", JSON.stringify(result));
+      history.push('/dashboard');
+    } catch (error) {
+      setError('Invalid email or password. Please try again.');
+    }
+  };
```

Comment:
Update the code to use `useNavigate` instead of `useHistory`.
`useHistory` is not available in `react-router-dom` v6.x and onwards. Please ensure to test thoroughly after making this change.